### PR TITLE
Add Punctuation characters set

### PR DESCRIPTION
**Pull request by artemklevtsov**
Sometimes is useful to check or strip punctuation.

### CODE REVIEW

File: lib/pure/strutils.nim [hunk 1]
```diff
@@ -89,6 +89,9 @@ const
     ## All the characters that count as whitespace (space, tab, vertical tab,
     ## carriage return, new line, form feed).
 
+  Punctuation* = {'\33'..'\47', '\58'..'\64', '\91'..'\96', '\123'..'\126'}
```
Comment (timotheecour) [inline lib/pure/strutils.nim]: why obfuscate with ord values, can't you instead use ',' etc?

File: lib/pure/strutils.nim [hunk 2]
```diff
@@ -89,6 +89,9 @@ const
...
+    ## All the characters that count as punctuation (comma, colon, dash etc).
```
Comment (timotheecour) [inline lib/pure/strutils.nim:93]: can you provide in this PR a link to what some other languages do here?

File: lib/pure/strutils.nim (hunk 1 above)
Comment (artemklevtsov) [reply lib/pure/strutils.nim]: Changed to symbols.

File: lib/pure/strutils.nim (hunk 2 above)
Comment (artemklevtsov) [reply lib/pure/strutils.nim:93]: For example:
- https://en.cppreference.com/w/c/string/byte/ispunct
- https://docs.python.org/3/library/string.html?highlight=string#string.punctuation

**Comment by artemklevtsov** [comment]
I also add the `isPunctAscii` function and some tests for it.

File: lib/pure/strutils.nim [hunk 3]
```diff
@@ -89,6 +89,9 @@ const
     ## All the characters that count as whitespace (space, tab, vertical tab,
     ## carriage return, new line, form feed).
 
+  Punctuation* = {'!', '"', '#', '$', '%', '&', '\'', '(', ')', '*', '+', ',', '-', '.', '/', ':', ';', '<', '=', '>', '?', '@', '[', '\\', ']', '^', '_', '`', '{', '|', '}', '~'}
+    ## All the characters that count as punctuation (comma, colon, dash etc).
+
```
Comment (timotheecour) [inline lib/pure/strutils.nim:94]: optional: how about also adding `Printable`?
refs https://docs.python.org/3/library/string.html?highlight=string#string.printable

could also be another PR

File: tests/stdlib/tstrutils.nim [hunk 4]
```diff
@@ -661,6 +661,13 @@ bar
     doAssert isSpaceAscii('\l')
     doAssert(not isSpaceAscii('A'))
 
+  block: # isPunctAscii
+    doAssert isPunctAscii(',')
+    doAssert isPunctAscii('.')
+    doAssert isPunctAscii(':')
+    doAssert isPunctAscii(';')
+    doAssert(not isPunctAscii('A'))
+
```
Comment (timotheecour) [inline tests/stdlib/tstrutils.nim]: and add `from std/setutils import toSet` on top

```suggestion
    # check this matches https://en.cppreference.com/w/c/string/byte/ispunct
    doAssert "!\"#$%&'()*+,-./:;<=>?@[\\]^_`{|}~".toSet == Punctuation

```

File: lib/pure/strutils.nim [hunk 5]
```diff
@@ -158,6 +161,14 @@ func isSpaceAscii*(c: char): bool {.rtl, extern: "nsuIsSpaceAsciiChar".} =
     doAssert isSpaceAscii('\t') == true
   return c in Whitespace
 
+func isPunctAscii*(c: char): bool {.rtl, extern: "nsuIsPunctAsciiChar".} =
```
Comment (timotheecour) [inline lib/pure/strutils.nim]: i don't think we need this proc, `c in Punctuation` seems good enough

Comment (artemklevtsov) [reply lib/pure/strutils.nim]: Ok. I'll remove it.

**Review by timotheecour — APPROVED** [review]

**Comment by Araq** [comment]
Seems more dangerous than useful, sorry. What does count as "punctuation" depends too much on the concrete use case. If you follow some spec, better write the set down yourself rather than relying on strutils's definition of it. And if you don't have a spec and use strutils.Punctuation, either we can never change its definition ("improve it because then it's consistent with Python...") or we break your code when we do.

**Comment by timotheecour** [comment]
=> followup https://github.com/nim-lang/Nim/pull/18193#discussion_r646161383
